### PR TITLE
feat(parser): class public/private fields are in ecma spec now

### PR DESCRIPTION
**Pull request by 3cp**
BREAKING CHANGE: move class public/private fields out of options.next

Also fixed a bug that prevented rejecting "class A { get x }".

### CODE REVIEW

File: test/test262-parser-tests/parser-tests.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ const skipped = {
     '79f882da06f88c9f.js',
     '92b6af54adef3624.js',
 
+    // class fields
+    '98204d734f8c72b3.js',
+    'ef81b93cf9bdb4ec.js',
```
Comment (3cp) [inline test/test262-parser-tests/parser-tests.ts:24]: test262-parser-tests seems quite out of date. The two files contain perfect valid JS code with class property definition.

```js
// 98204d734f8c72b3.js
(class {a})
// ef81b93cf9bdb4ec.js
(class {a=0})
```

**Comment by 3cp** [comment]
@fisker do you know the relation between [test262-parser-tests](https://github.com/tc39/test262-parser-tests) and [test262](https://github.com/tc39/test262)?

The test262-parser-tests we use seems quite out of date.

**Comment by 3cp** [comment]
Our `options.next` is seriously behind current ecmascript spec. This is a beginning PR that I want to move some features out of `next` flag.

File: src/parser.ts [hunk 2]
```diff
@@ -8620,14 +8620,29 @@ export function parsePropertyDefinition(
...
+          decorators
+        }
+      : {
+          type: 'PropertyDefinition',
+          key,
+          value,
+          static: (state & PropertyKind.Static) > 0,
+          computed: (state & PropertyKind.Computed) > 0
+        }) as any
+  );
```
Comment (fisker) [inline src/parser.ts:8645]: These two branches are quite big, can we use 

```js
decorator: (context & Context.OptionsNext ? decorator : undefined)
```

?

Or

```js
...(context & Context.OptionsNext ? {decorator} : undefined)
``

if tge key existence matters.

**Comment by fisker** [comment]
> @fisker do you know the relation between [test262-parser-tests](https://github.com/tc39/test262-parser-tests) and [test262](https://github.com/tc39/test262)?
> 
> The test262-parser-tests we use seems quite out of date.

Didn't involved in parser much, I'm not familiar with them.

File: src/parser.ts (hunk 2 above)
Comment (3cp) [reply src/parser.ts:8645]: There are few cases like this. I will clean them up.

**Review by fisker — APPROVED** [review]

**Pull request merged**